hera_cal/io.py

```diff
@@ -1685,3 +1687,121 @@ def update_cal(infilename, outfilename, gains=None, flags=None, quals=None, add_
 
     # Write to calfits file
     cal.write_calfits(outfilename, clobber=clobber)
+
+
+def int_rep_test(test_string):
+    """
+    Test whether a string represents an integer.
+
+    Arguments
+    ---------
+        test_string, string. String to be tested.
+    Returns
+    ---------
+        boolean. True if string can be represented by an integer and False otherwise. 
+    """
+    try:
+        test_int = int(test_string)
+        return True
+    except ValueError:
+        return False
+
+
+def _parse_antpairpol_list_string(antpairpol_list_string):
+    """
+    Helper method for parsing user provided lists of baselines.
+
+    Arguments
+    ---------
+        baseline_list_string: string
+            list of baselines provided as a string
+            that is a semi-colon separated list of antenna-pol triplets.
+            white-space (tabs, spaces, newlines) is ignored.
+            e.g. 1, 2,ee;0,0,nn; 0, 10,en
+    Returns
+    -------
+        A list of baseline antpairpol tuples
+    """
+    assert isinstance(antpairpol_list_string, str), "Must provide a string to be parsed!"
+    # strip spaces
+    antpairpol_list_string = "".join(antpairpol_list_string.split())
+    # split on semi-colons
+    antpairpol_list = antpairpol_list_string.split(";")
+    for appnum, antpp in enumerate(antpairpol_list):
+        antpp = antpp.split(",")
+        if not len(antpp) == 3:
+            raise AssertionError("Must provide semicolon separated list of comma separated triplet antpairpols!")
+        if not int_rep_test(antpp[0]):
+            raise AssertionError("invalid antenna number %s in %s provided!" % (antpp[0], str(antpp)))
+        if not int_rep_test(antpp[1]):
+            raise AssertionError("invalid antenna number %s in %s provided!" % (antpp[1], str(antpp)))
+        # check valid baseline string
+        if not antpp[2] in POL_STR2NUM_DICT and not antpp[2] in ['ee', 'nn', 'en', 'ne']:
+            raise AssertionError("invalid polarization, %s provided!" % antpp[2])
+        antpairpol = (int(antpp[0]), int(antpp[1]), antpp[2])
+        antpairpol_list[appnum] = antpairpol
+    return antpairpol_list
+
+
+def generate_antpairpol_parallelization_files(filename, writedir, polarizations=None, bls_per_chunk=1):
+    """
+    Generate text files containing lists of basteline-pol-pairs
+
+    Arguments
+    ---------
+        filename: string
+            name of file whose baselines we will generate parallization files for.
+        writedir: string
+            name of directory to store parallelization file.
+        polarizations: list optional,
+            polarizations to include. default is all polarizations in the data file.
+        bls_per_chunk: integer optional,
+            number of baselines to include in each parallelized chunk.
+            default is 1.
+    Returns
+    -------
+        Nothing, just writes files with antpolpair lists for parallelization.
+    """
+    # load in data file
```

Comment:
Since this function loads no data, it seems like it would be trivially fast. Does it not make more sense to just have each job take an index and let each job analyze the same file, figure out the baselines, and figure out which of them it should analyze? I'm generally against lots of messy file creation (especially in RTP).

Reply:
Yes. This lets us use RTP infrastructure (instead of writing a bunch of new functionality).